samples/boards/nrf52/mesh/onoff_level_app/src/mesh/device_composition.c

```diff
@@ -0,0 +1,340 @@
+#include "common.h"
+
+static struct bt_mesh_cfg_srv cfg_srv = {
+	.relay = BT_MESH_RELAY_DISABLED,
+	.beacon = BT_MESH_BEACON_ENABLED,
+
+#if defined(CONFIG_BT_MESH_FRIEND)
+	.frnd = BT_MESH_FRIEND_ENABLED,
+#else
+	.frnd = BT_MESH_FRIEND_NOT_SUPPORTED,
+#endif
+
+#if defined(CONFIG_BT_MESH_GATT_PROXY)
+	.gatt_proxy = BT_MESH_GATT_PROXY_ENABLED,
+#else
+	.gatt_proxy = BT_MESH_GATT_PROXY_NOT_SUPPORTED,
+#endif
+
+	.default_ttl = 7,
+
+	/* 3 transmissions with 20ms interval */
+	.net_transmit = BT_MESH_TRANSMIT(2, 20),
+	.relay_retransmit = BT_MESH_TRANSMIT(2, 20),
+};
+
+static struct bt_mesh_health_srv health_srv = {
+};
+
+BT_MESH_HEALTH_PUB_DEFINE(health_pub, 0);
+
+BT_MESH_MODEL_PUB_DEFINE(gen_onoff_srv_pub_root, NULL, 2 + 2);
+BT_MESH_MODEL_PUB_DEFINE(gen_onoff_cli_pub_root, NULL, 2 + 2);
+BT_MESH_MODEL_PUB_DEFINE(gen_level_srv_pub_root, NULL, 2 + 2);
+BT_MESH_MODEL_PUB_DEFINE(gen_level_cli_pub_root, NULL, 2 + 2 + 2);
+
+BT_MESH_MODEL_PUB_DEFINE(gen_onoff_srv_pub_s0, NULL, 2 + 2);
+BT_MESH_MODEL_PUB_DEFINE(gen_onoff_cli_pub_s0, NULL, 2 + 2);
+
+struct server_state   gen_onoff_srv_user_data_root = {
+	.previous = 0xFFFFFFFF, /* it should be anything except {0,1} */
+	.model_instance = 1,
+};
+
+struct server_state   gen_level_srv_user_data_root = {
+
+	.previous = 0xFFFFFFFF,	/* it should be anything
+				 * except {-32768 to 32767}
+				*/
+	.model_instance = 1,
+};
+
+struct server_state   gen_onoff_srv_user_data_s0 = {
+	.previous = 0xFFFFFFFF,	/* it should be anything except {0,1} */
+	.model_instance = 2,
+};
+
+/* ----------------------- message handlers (Start) ----------------------- */
+
+static void gen_onoff_get(struct bt_mesh_model *model,
+			  struct bt_mesh_msg_ctx *ctx,
+			  struct net_buf_simple *buf)
+{
+	struct net_buf_simple *msg = NET_BUF_SIMPLE(2 + 1 + 4);
+	struct server_state *state = model->user_data;
+
+	bt_mesh_model_msg_init(msg, BT_MESH_MODEL_OP_2(0x82, 0x04));
+
+	net_buf_simple_add_u8(msg, state->current);
+
+	if (bt_mesh_model_send(model, ctx, msg, NULL, NULL)) {
+		printk("Unable to send ONOFF_SRV Status response\n");
+	}
+}
+
+static void gen_onoff_set_unack(struct bt_mesh_model *model,
+			  struct bt_mesh_msg_ctx *ctx,
+			  struct net_buf_simple *buf)
+{
+	int err = 0;
```

Comment:
I'd move this variable inside the smallest scope where it's used, i.e. to the beginning of the `if (model->pub->addr != BT_MESH_ADDR_UNASSIGNED)` branch. Also, there's no need to initialize it to zero.